CONTRIBUTING.md

```diff
@@ -39,11 +40,33 @@ stack test
 stack bench
 ```
 
+
 ## Troubleshooting
 
-- If you're using Stack, make sure you have version >= 1.6.1 ([1], [2])
+- If you're using Stack, make sure you have version >= 1.6.1
+  ([stack#3524](https://github.com/commercialhaskell/stack/issues/3624),
+  [stack#3345](https://github.com/commercialhaskell/stack/issues/3345)).
+
+
+## Sending Pull Requests
+
+When you send a pull request, please:
+
+- Link to libraries@haskell.org discussion thread if you are changing the public
+  API.
 
+- If you are requesting a change that is likely to affect performance, we will
+  be able to evaluate it better if you include the results of running the
+  benchmarks before and after. If the current benchmarks cannot demonstrate
+  a desired difference, please try to add one or more new benchmarks to do so.
+  If there are significant changes, please include the benchmark results in
+  your commit message.
 
-[1] https://github.com/commercialhaskell/stack/issues/3624
+- If you are requesting a change that adds new functionality or affects
+  behaviour, please add QuickCheck properties exercising the code if they
+  do not already exist. If the problem occurs too rarely for QuickCheck to
+  hit then consider adding unit tests as well.
+  
+- Update change log for non-trivial changes.
 
-[2] https://github.com/commercialhaskell/stack/issues/3345
+- Let us know how you wish to be credited in the changelog.
```

Comment:
Perhaps "Please let us know"?

Reply:
Changed, although just realized this looks a little strange because there is a "please:" before the list as well. Shall I revert this one?